lib/src/Uncertainty/Distribution/UniformOrderStatistics.cxx

```diff
@@ -0,0 +1,470 @@
+{
+  if (q.getDimension() != dimension_) throw InvalidArgumentException(HERE) << "Error: cannot compute a sequential conditional PDF at a quantile level vector of dimension=" << q.getDimension() << " not equal to the distribution dimension=" << dimension_;
+  if ((q[0] < 0.0) || (q[0] > 1.0)) throw InvalidArgumentException(HERE) << "Error: cannot compute a conditional quantile for a probability level outside of [0, 1]";
+  Point result(dimension_);
+  result[0] = 1.0 - std::pow(1.0 - q[0], 1.0 / dimension_);
+  for (UnsignedInteger k = 1; k < dimension_; ++k)
+    {
+      if ((q[k] < 0.0) || (q[k] > 1.0)) throw InvalidArgumentException(HERE) << "Error: cannot compute a conditional quantile for a probability level outside of [0, 1]";
+      result[k] = 1.0 - (1.0 - result[k - 1]) * std::pow(1.0 - q[k], 1.0 / (dimension_ - k));
+    }
+  return result;
+}
+
+/* Compute the mean of the distribution */
+void UniformOrderStatistics::computeMean() const
+{
+  mean_ = Point(dimension_);
+  for (UnsignedInteger i = 0; i < dimension_; ++i)
+    mean_[i] =  (i + 1.0) / (dimension_ + 1.0);
+  isAlreadyComputedMean_ = true;
+}
+
+/* Compute the entropy of the distribution */
+Scalar UniformOrderStatistics::computeEntropy() const
+{
+  return -SpecFunc::LogGamma(dimension_ + 1.0);
+}
+
+/* Get the standard deviation of the distribution */
+Point UniformOrderStatistics::getStandardDeviation() const
+{
+  Point sigma(dimension_);
+  for (UnsignedInteger i = 0; i < dimension_; ++i)
+    sigma[i] = 1.0 / (dimension_ + 1.0) * std::sqrt((i + 1.0) * (dimension_ - i) / (dimension_ + 2.0));
+  return sigma;
+}
+
+/* Get the skewness of the distribution */
+Point UniformOrderStatistics::getSkewness() const
+{
+  Point skewness(dimension_);
+  for (UnsignedInteger i = 0; i < dimension_; ++i)
+    skewness[i] = 2.0 * (dimension_ - 1.0 - 2.0 * i) / (dimension_ + 3.0) * std::sqrt((dimension_ + 2.0) / ((i + 1.0) * (dimension_ - i)));
+  return skewness;
+}
+
+/* Get the kurtosis of the distribution */
+Point UniformOrderStatistics::getKurtosis() const
+{
+  Point kurtosis(dimension_);
+  for (UnsignedInteger i = 0; i < dimension_; ++i)
+    kurtosis[i] = 3.0 * (dimension_ + 2.0) * (2.0 * (dimension_ + 1.0) * (dimension_ + 1.0) + (i + 1.0) * ((dimension_ + 1.0) - 6.0) * (dimension_ - i)) / ((i + 1.0) * (dimension_ - i) * (3.0 + (dimension_ + 1.0)) * (2.0 + (dimension_ + 1.0)));
+  return kurtosis;
+}
+
+/* Compute the covariance of the distribution */
+void UniformOrderStatistics::computeCovariance() const
+{
+  DistributionImplementation::computeCovarianceGeneral();
+  isAlreadyComputedCovariance_ = true;
+}
+
+/* Get the quantile of the distribution */
+Scalar UniformOrderStatistics::computeScalarQuantile(const Scalar prob,
+    const Bool tail) const
+{
+  if (tail) return 1.0 - prob;
+  return prob;
+}
+
+/* Get the i-th marginal distribution */
+Distribution UniformOrderStatistics::getMarginal(const UnsignedInteger i) const
```

Comment:
According to (Arnold et al., 2008) eq. 2.2.15 page 13, this is a Beta(i, n - i + 1), isn'it?

Reply:
No. You have to take into account the shift of 1 between the C++/Python indexing and the textbook indexing.

Reply:
Sorry, my comment is not relevant. I did not carefully read the `getMarginal(Indices(1, i))`: this triggers the code below which takes into account this particular case.